docs/en/integrations/language-clients/rust.md

```diff
@@ -0,0 +1,517 @@
+* `uuid` — adds `serde::uuid` to work with [uuid](https://docs.rs/uuid) crate.
+* `time` — adds `serde::time` to work with [time](https://docs.rs/time) crate.
+
+:::important
+When connecting to ClickHouse via an `HTTPS` url, either the `native-tls` or `rustls-tls` feature should be enabled.
+If both are enabled, the `rustls-tls` feature will take precedence.
+:::
+
+## ClickHouse versions compatibility
+
+The client is compatible with the LTS or newer versions of ClickHouse, as well as ClickHouse Cloud.
+
+ClickHouse server older than v22.6 handles RowBinary [incorrectly in some rare cases](https://github.com/ClickHouse/ClickHouse/issues/37420). 
+You could use v0.11+ and enable `wa-37420` feature to solve this problem. Note: this feature should not be used with newer ClickHouse versions.
+
+## Examples
+
+We aim to cover various scenarios of client usage with the [examples](https://github.com/ClickHouse/clickhouse-rs/blob/main/examples) in the client repository. The overview is available in the [examples README](https://github.com/ClickHouse/clickhouse-rs/blob/main/examples/README.md#overview).
+
+If something is unclear or missing from the examples or from the following documentation, feel free to [contact us](./rust.md#contact-us).
+
+## Usage
+
+:::note
+[ch2rs](https://github.com/ClickHouse/ch2rs) crate is useful to generate a row type from ClickHouse.
+:::
+
+### Creating a client instance
+
+```rust
+use clickhouse::Client;
+
+let client = Client::default()
+    // should include both protocol and port
+    .with_url("http://localhost:8123")
+    .with_user("name")
+    .with_password("123")
+    .with_database("test");
+```
+
+:::tip
+Reuse created clients or clone them in order to reuse the underlying hyper connection pool.
+:::
+
+### HTTPS or ClickHouse Cloud connection
+
+HTTPS works with either `rustls-tls` or `native-tls` cargo features.
+
+Then, create client as usual. In this example, the environment variables are used to store the connection details:
+
+```rust
+fn read_env_var(key: &str) -> String {
+    env::var(key).unwrap_or_else(|_| panic!("{key} env variable should be set"))
+}
+
+let client = Client::default()
+    .with_url(read_env_var("CLICKHOUSE_URL"))
+    .with_user(read_env_var("CLICKHOUSE_USER"))
+    .with_password(read_env_var("CLICKHOUSE_PASSWORD"));
+```
+
+NB: the URL should include both protocol and port, e.g. `https://instance.clickhouse.cloud:8443`.
+
+See also: 
+- [HTTPS with ClickHouse Cloud example](https://github.com/ClickHouse/clickhouse-rs/blob/main/examples/clickhouse_cloud.rs) in the client repo. This should be applicable to on-premise HTTPS connections as well.
+
+### Selecting rows
+
+```rust
+use serde::Deserialize;
+use clickhouse::Row;
+use clickhouse::sql::Identifier;
+
+#[derive(Row, Deserialize)]
+struct MyRow<'a> {
+    no: u32,
+    name: &'a str,
+}
+
+let table_name = "some";
+let mut cursor = client
+    .query("SELECT ?fields FROM ? WHERE no BETWEEN ? AND ?")
```

Comment:
Does the client support server-side bindings?

Reply:
Not as of now.

Reply:
https://github.com/ClickHouse/clickhouse-rs/issues/142